src/encoded/static/components/collection.js

```diff
@@ -109,22 +141,58 @@ class FacetChart extends React.Component {
 
     render() {
         const { chartId, facet } = this.props;
+        let terms = [];
+        let otherTerms;
+
+        // If we have more facet terms than we allow to be displayed (determined through
+        // MAX_CHART_ITEMS), then just get MAX_CHART_ITEMS to display as usual, and collect the
+        // rest to display as "other."
+        if (facet.terms.length > MAX_CHART_ITEMS) {
+            // We have more facet terms than we display individually. First get the terms to
+            // display as individual items.
+            terms = facet.terms.slice(0, MAX_CHART_ITEMS);
+
+            // Now collect the terms to display as "other."
+            otherTerms = facet.terms.slice(MAX_CHART_ITEMS);
+        } else {
+            // We don't have enough facet terms for an "other" category, so just use the whole
+            // array of facet terms.
+            terms = facet.terms;
+            otherTerms = [];
+        }
 
         // Extract the arrays of labels from the facet keys, and the arrays of corresponding counts
         // from the facet doc_counts. Only use non-zero facet terms in the charts. IF we have no
         // usable data, both these arrays have no entries. componentDidMount assumes these arrays
         // have been populated.
         this.values = [];
         this.labels = [];
-        facet.terms.forEach((term) => {
+        terms.forEach((term) => {
             if (term.doc_count) {
```

Comment:
Hmm, so if there are terms with no doc_count we can end up with a chart that has fewer than MAX_CHART_ITEMS wedges but still has an "other"...